### PR TITLE
Convert modal to jsx

### PR DESCRIPTION
**Pull request by nyrosmith**
Associated Issue: #3506 
### Summary of Changes
Changed Modal to JSX syntax

I am quite unsure about the "Slide"-component...

### CODE REVIEW

File: src/components/shared/Modal.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 // @flow
 
-import { DOM as dom, createFactory, Component, PropTypes } from "react";
+import React, { PropTypes, Component } from "react";
 import type { Children } from "react";
 import classnames from "classnames";
-import _Transition from "react-transition-group/Transition";
-const Transition = createFactory(_Transition);
-
+import Transition from "react-transition-group/Transition";
```
Comment (jasonLaster) [inline src/components/shared/Modal.js]: is this used?

Comment (nyrosmith) [inline src/components/shared/Modal.js]: I am really unsure.
Transition is used within Slide. Which is the default export.
I did a quick search but couldn't find any usage of Slide, but bc I am a total noob on your codebase this doesnot mean I didn't oversee something

**Comment by nyrosmith** [comment]
I could comment out the Slide function and See if it breaks the bundle. But this would only work if you do not dynamics require()...

File: src/components/shared/Modal.js (hunk 1 above)
Comment (wldcordeiro) [inline src/components/shared/Modal.js]: Slide is a component. You can convert the default export to JSX and make this proper change.

Comment (wldcordeiro) [inline src/components/shared/Modal.js]: Something like this:

```jsx
export default function Slide({
  in: inProp,
  children,
  handleClose
}: SlideProps) {
  return (
    <Transition id={inProp} timeout={175} appear>
      {status => <Modal status={status} handleClose={handleClose}>{children}</Modal>}
    </Transition>
  );
}
```

**Comment by nyrosmith** [comment]
@wldcordeiro  @jasonLaster Updated based on your suggestions.
Thx for guidance!
Sorry if formatting doesnot fit but I am unable to run prettier...

**Pull request merged**